### PR TITLE
Fix/cleanup packages (3959)

### PR DESCRIPTION
**Pull request by marcussc**
## Description

If the App got killed during a Risk calculation, it could happen that the written KeyPackages would remain on Disk. 
This PR addresses this Issue. Old KeyPackages now get cleaned up before new KeyPackages get written.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-3959

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		try createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+		Log.debug("Created: \(url)", log: .localData)
+		return url
+	}
+	
+	// MARK: - Private
+	
+	private func removeOldKeypackages() {
+		Log.debug("Removing \(keyPackageURL())", log: .localData)
+		try? removeItem(at: keyPackageURL()) // Delete the whole dir to make sure no old data is left behind
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift:27]: Would be nice if we could log if an error happens here.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift [hunk 2]
```diff
@@ -11,10 +11,16 @@ struct WrittenPackages {
...
-			try? fileManager.removeItem(at: url)
+		guard let directoryURL = urls.first?.deletingLastPathComponent() else {
+			return
 		}
+		
+		let fileManager = FileManager()
+		Log.info("Removing: \(directoryURL)", log: .localData)
+		
+		// Remove the whole directory, instead of removing each file and than forget to remove the directory
+		try? fileManager.removeItem(at: directoryURL)
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift:22]: Could you use removeOldKeypackages here?

**Review by ArturFriesen — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift]: Added.

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift (hunk 2 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/AppleFilesWriter.swift:22]: I would not do this, `removeOldKeypackages` is only a fallback and `AppleFilesWriter` handles Keypackages one at a time.

File: src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift [hunk 3]
```diff
@@ -0,0 +1,55 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+/// This extension handles file clean up before new files get written to disk.
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift:7]:
```suggestion
/// This extension handles creating a key package directory and file clean up before new files get written to disk.
```

File: src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+		
+		let url = keyPackageURL().appendingPathComponent(UUID().uuidString, isDirectory: true)
+		try createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+		Log.debug("Created: \(url)", log: .localData)
+		return url
+	}
+	
+	// MARK: - Private
+	
+	private func removeOldKeypackages() {
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Extensions/FileManager+KeyPackageStorage.swift:25]:
```suggestion
	private func removeOldKeyPackages() {
```

**Review by nickguendling — APPROVED** [review]

**Pull request merged**